corehq/ex-submodules/couchexport/writers.py

```diff
@@ -177,8 +178,11 @@ def open(self, header_table, file, max_column_size=2000, table_titles=None, arch
 
     def add_table(self, table_index, headers, table_title=None):
         def _clean_name(name):
-            if not isinstance(name, six.text_type):
+            if isinstance(name, six.binary_type):
                 name = name.decode('utf8')
+            elif isinstance(name, Promise):
+                # noinspection PyCompatibility
+                name = unicode(name) if six.PY2 else str(name)
```

Comment:
Could you change these too ```six.text_type``` and ```six.binary_type```?

Reply:
No. I tried, but `six.text_type` doesn't accept [Django Promise subclasses](https://docs.djangoproject.com/en/1.11/ref/unicode/#translated-strings). It throws
```python
TypeError: coercing to Unicode: need string or buffer, __proxy__ found
```

The alternative is to call a private method with ...
```python
    elif isinstance(name, Promise):
        name = name._proxy____text_cast()
```
... but I thought that was even worse than this.